File: scripts/strip_emojis.js
const fs = require('fs');
const path = require('path');

function stripEmojis(filePath) {
    let content = fs.readFileSync(filePath, 'utf8');
    
    // 1. Preserve important injury markers
    content = content.replace(/🟢/g, '__GREEN_CIRCLE__');
    content = content.replace(/🟡/g, '__YELLOW_CIRCLE__');
    content = content.replace(/🔴/g, '__RED_CIRCLE__');
    content = content.replace(/⛔/g, '__NO_ENTRY__');

    // Using Unicode property escapes to match emojis. 
    const emojiRegex = /\p{Emoji_Presentation}/gu;
    
    // Also remove some specific symbols that might not fall strictly under Emoji_Presentation but are used as icons
    const otherSymbols = /[🧘🔥⚡⏸⇄💡🚀🏆📈🛡💪🎯🥗🙂⚠️✓×]/gu;

    let newContent = content.replace(emojiRegex, '').replace(otherSymbols, '');
    
    // Fix up some common double spaces left after removing emojis
    newContent = newContent.replace(/  +/g, ' ').replace(/ \n/g, '\n');

    // 2. Restore important injury markers
    newContent = newContent.replace(/__GREEN_CIRCLE__/g, '🟢');
    newContent = newContent.replace(/__YELLOW_CIRCLE__/g, '🟡');
    newContent = newContent.replace(/__RED_CIRCLE__/g, '🔴');
    newContent = newContent.replace(/__NO_ENTRY__/g, '⛔');

    if (newContent !== content) {
        fs.writeFileSync(filePath, newContent, 'utf8');
        console.log(`Stripped emojis from: ${filePath}`);
        return 1;
    }
    return 0;
}

function walkSync(dir, filelist = []) {
    fs.readdirSync(dir).forEach(file => {
        const dirFile = path.join(dir, file);
        try {
            filelist = walkSync(dirFile, filelist);
        } catch (err) {
            if (err.code === 'ENOTDIR' || err.code === 'EBADF') {
                if (file.endsWith('.html') || file.endsWith('.js') || file.endsWith('.json')) {
                    filelist.push(dirFile);
                }
            }
        }
    });
    return filelist;
}

const baseDir = path.join('e:', 'MoS', 'website');
const files = walkSync(baseDir);
let modified = 0;

files.forEach(f => {
    modified += stripEmojis(f);
});

console.log(`Finished. Modified ${modified} files.`);

File: scripts/strip_emojis.py
import os
import re
import sys

# Broad regex for emojis
emoji_pattern = re.compile(
    u'('
    u'\U0001F600-\U0001F64F'  # emoticons
    u'|\U0001F300-\U0001F5FF'  # symbols & pictographs
    u'|\U0001F680-\U0001F6FF'  # transport & map symbols
    u'|\U0001F700-\U0001F77F'  # alchemical symbols
    u'|\U0001F780-\U0001F7FF'  # Geometric Shapes Extended
    u'|\U0001F800-\U0001F8FF'  # Supplemental Arrows-C
    u'|\U0001F900-\U0001F9FF'  # Supplemental Symbols and Pictographs
    u'|\U0001FA00-\U0001FA6F'  # Chess Symbols
    u'|\U0001FA70-\U0001FAFF'  # Symbols and Pictographs Extended-A
    u'|\u2600-\u26FF'          # miscellaneous symbols
    u'|\u2700-\u27BF'          # dingbats
    u'|\u231a-\u231b|\u23e9-\u23ec|\u23f0|\u23f3' # Some watches/clocks
    u'|\u2B50|\u2B55'          # star/circle
    u')+', re.UNICODE)

base_dir = r'e:\MoS\website'
total_stripped = 0

for root, dirs, files in os.walk(base_dir):
    for f in files:
        if f.endswith(('.html', '.js', '.css', '.json')):
            filepath = os.path.join(root, f)
            with open(filepath, 'r', encoding='utf-8', errors='ignore') as file:
                content = file.read()
            
            # Remove emojis
            new_content = emoji_pattern.sub('', content)
            
            if new_content != content:
                print('Stripped emojis from:', filepath)
                with open(filepath, 'w', encoding='utf-8', newline='') as file:
                    file.write(new_content)
                total_stripped += 1

print(f"Finished. Modified {total_stripped} files.")
